Argo/Functions/decode.swift

```diff
@@ -13,3 +15,21 @@ public func decode<T: Decodable where T == T.DecodedType>(object: AnyObject) ->
 public func decode<T: Decodable where T == T.DecodedType>(object: AnyObject) -> Decoded<[T]> {
   return Array<T>.decode(JSON.parse(object))
 }
+
+// MARK: decode with a root key
+
+public func decode<T: Decodable where T == T.DecodedType>(rootKey: String, _ object: AnyObject) -> T? {
```

Comment:
I agree, I initially had it that way but I switched it up for partial application purposes. Check out the tests that changed in this PR. It's nice to be able to partially apply the root key then pass in the `AnyObject`.